python/lancedb/table.py

```diff
@@ -373,15 +375,21 @@ def restore(self, version: int):
         3
         """
         max_ver = max([v["version"] for v in self._dataset.versions()])
+        if version is None:
+            version = self.version
+
         if version < 1 or version >= max_ver:
```

Comment:
Do you intend for this to be `>=`?  Isn't `version == max_ver` supposed to be a no-op (and not an error)?

Reply:
good catch let me add a unit test